File: interviewquestions/ispalindrome/recursive-palindrome.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
    // Convert number to string
    x = x.toString();

    // Define the recursive function
    var recursive = function (x, low, high) {
        if (high <= low) {
            return true; // Base case: all characters checked
        } else if (x.charAt(low) != x.charAt(high)) {
            return false; // Base case: characters don't match
        } else {
            return recursive(x, low + 1, high - 1); // Recursive call
        }
    };

    // Call the recursive function with initial values
    return recursive(x, 0, x.length - 1);
};

// Example usage:
console.log(isPalindrome(121)); // Output: true
console.log(isPalindrome(-121)); // Output: false
console.log(isPalindrome(10)); // Output: false

File: interviewquestions/ispalindrome/RecursivePalindrome.java
package interviewquestions.ispalindrome;

public class RecursivePalindrome {
    public static boolean isPalinDrome(String s){
        return isPalinDrome(s, 0 , s.length() - 1);
    }
    
    public static boolean isPalinDrome(String s, int low, int high){ //more efficent than creating substring
        if (high <= low) 
            return true;
        else if (s.charAt(low) != s.charAt(high)) //base case
            return false;
        else 
            return isPalinDrome(s, low + 1, high - 1); //check next value
    }
}